src/domain/products/product-form/form/product-form-context.tsx

```diff
@@ -66,6 +71,15 @@ export const ProductFormProvider = ({
   const [variants, setVariants] = React.useState<any[]>([])
   const [productOptions, setProductOptions] = React.useState<any[]>([])
   const [hasImagesChanged, setHasImagesChanged] = React.useState(false)
+  const [hasSalesChannelsChanged, setHasSalesChannelsChanged] = React.useState(
+    false
+  )
+
+  // SALES CHANNELS
+  const { isFeatureEnabled } = React.useContext(FeatureFlagContext)
+  const [salesChannels, setSalesChannels] = React.useState<any[]>(
+    product.sales_channels
```

Comment:
suggestion: should we populate the default state directly by taking into account the flag? such as:
```ts
  const [salesChannels, setSalesChannels] = React.useState<any[]>(
    isFeatureEnabled("sales_channels") ? product.sales_channels : []
  )
```

Reply:
suggestion: should we replace `any[]` by `SalesChannel[]`